### PR TITLE
docs: Root access is insecure: draft for expanded security admin topics

### PR DESCRIPTION
**Pull request by lsgunn-teleport**
There are a lot of topics covered in https://github.com/gravitational/teleport.e/issues/1359.
This draft covers the first set:
- [x] Agent root access is insecure
- [x] Elevated access to auto discovery
- [x] Elevated access to auth or proxy server
- [x] Storage security requirements (probably needs to live somewhere else)
- [x] Network security requirements (probably needs to live somewhere else)

They don't fit together perfectly or under this page title, but it's a starting point.

### CODE REVIEW

File: docs/pages/management/security/privilege-elevation.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+To prevent changes to labels from granting elevated privileges, you should:
+
+- Carefully consider how allow and deny rules are defined and interact in every role.
+- Restrict access to the Teleport process and Teleport configuration files on all nodes in the cluster.
+- Use care in defining dynamic labels that execute commands.
+- Restrict access to any command that may be run as part of dynamic labels.
+- Follow the **principle of least privilege** in creating and assigning roles. 
+  Don't give users permissive roles when more a restrictive role will do. For example, don't 
+  assign users the preset `access,editor` roles, which give them permission to access and edit
```
Comment (zmb3) [inline docs/pages/management/security/privilege-elevation.mdx:28]:
```suggestion
  assign users the preset `access` or `editor` roles, which give them permission to access and edit 
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/management/security.mdx [hunk 2]
```diff
@@ -1,11 +1,26 @@
 ---
-title: Security
-description: Hardening the security of your Teleport deployment
-layout: tocless-doc
+title: Secure cluster operations
```
Comment (ptgott) [inline docs/pages/management/security.mdx:2]:
```suggestion
title: Harden your Teleport cluster
```
Suggested to make this more specific:
- It's advice for making your Teleport cluster more secure, not operating securely within a cluster
- It makes it more explicit that we're referring to Teleport clusters for users arriving from Google

**Review by ptgott — APPROVED** [review]

File: docs/pages/management/security.mdx [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+convenience and security that you should take into consideration.
+
+The topics in this section are intended to help secure your Teleport cluster in a way that's acceptable and 
+reasonable for your organization.
+
+You should note that the security practices covered in this section aren't necessarily reflected in the 
+examples used in the documentation. Examples in the documentation are primarily intended for demonstration 
+purposes and for development environments.
+
+- [Privilege Elevation](./security/privilege-elevation.mdx). Learn about potential risks of allowing privileged
```
Comment (jentfoo) [inline docs/pages/management/security.mdx:18]: `Privilege Elevation` sounds like you are elevating your privileges beyond what was assigned.  I think `Elevated Privileges` or `Sensitive Access` or `Access Risks` sound more accurate.

File: docs/pages/management/security/privilege-elevation.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+for granting or denying access to critical resources.
+Users who can add or modify labels can—either intentionally or unintentionally—make changes 
+that result in privilege elevation when other users' roles are evaluated. 
+
+To prevent changes to labels from granting elevated privileges, you should:
+
+- Carefully consider how allow and deny rules are defined and interact in every role.
+- Restrict access to the Teleport process and Teleport configuration files on all nodes in the cluster.
+- Use care in defining dynamic labels that execute commands.
+- Restrict access to any command that may be run as part of dynamic labels.
```
Comment (jentfoo) [inline docs/pages/management/security/privilege-elevation.mdx:25]: I would reword this to something like:
` - Consider how dynamic labels are sourced, restrict possible modifications to the command or the data sources used`

File: docs/pages/management/security/privilege-elevation.mdx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+To prevent elevated privileges from resulting in security incidents, you should adhere to the 
+following guidelines:
+
+- Create new, non-root, users and use test instances for experimenting with Teleport.
+- Run all Teleport agent services except the SSH Service as a non-root user. 
+  The SSH Service requires root access, so you should limit access to the SSH Service. The Teleport Proxy
+  Service must also have root permission—or the `CAP_NET_BIND_SERVICE` capability—to allow Teleport listen 
+  on any port number less than 1024, including port 443.
+- Avoid using `sudo` in production environments unless it's necessary.
```
Comment (jentfoo) [inline docs/pages/management/security/privilege-elevation.mdx:51]: `Avoid allowing`
